### DISCUSSION TITLE
If statement in expression / virtual column

### DISCUSSION THREAD

**Discussion by abf7d** [vaexio/vaex#1804] [Q&A] [answered]
I want to set a virtual column to a calculation using another column in Vaex. I need to use an if statement inside this calculation. In general I want to call
```
df['calculation_col'] = log(df['original_col']) if df['original_col'] == 0 else -4
```
I then try to run the count function in Vaex:
```
hist = df.count(
        binby='calculation_col',
        limits=limits,
        shape=binnum,
        delay=True
    )
```
When I try to execute this code I get the error `ValueError: zero-size array to reduction operation minimum which has no identity`.

How can I use a conditional for a virtual column in Vaex?

**Comment by maartenbreddels** [comment, marked as answer]
Hi,

debug tip: If you print this out (in the repr) you'll probably see what is evaluates into
> `df['original_col']) if df['original_col'] == 0 else -4`

you can use where instead:

`(df['original_col']) == 0).where(df['original_col'], -4)`

Hope that helps

cheers,

Maarten

**Reply by abf7d** [reply]
This worked! Thank you so much!
